lib/ui/platform_dispatcher.dart

```diff
@@ -801,11 +801,46 @@ class PlatformDispatcher {
   ///
   ///  * [SchedulerBinding], the Flutter framework class which manages the
   ///    scheduling of frames.
+  ///  * [scheduleWarmUpFrame], which should only be used to schedule warm up
+  ///    frames.
   void scheduleFrame() => _scheduleFrame();
 
   @Native<Void Function()>(symbol: 'PlatformConfigurationNativeApi::ScheduleFrame')
   external static void _scheduleFrame();
 
+  /// Schedule a frame to run as soon as possible, rather than waiting for the
+  /// engine to request a frame in response to a system "Vsync" signal.
+  ///
+  /// This method is used during application startup so that the first frame
+  /// (which is likely to be quite expensive) can start a few extra milliseconds
+  /// earlier. Using it in other situations might lead to unintended results,
+  /// such as screen tearing. Depending on platforms and situations, the warm up
+  /// frame might or might not be actually rendered onto the screen.
+  ///
+  /// For more introduction to the warm up frame, see
+  /// [SchedulerBinding.scheduleWarmUpFrame].
+  ///
+  /// This method uses the provided callbacks as the begin frame callback and
+  /// the draw frame callback instead of [onBeginFrame] and [onDrawFrame].
+  ///
+  /// See also:
+  ///
+  ///  * [SchedulerBinding.scheduleWarmUpFrame], which uses this method, and
+  ///    introduces the warm up frame in more details.
+  ///  * [scheduleFrame], which schedules the frame at the next appropriate
+  ///    opportunity and should be used to render regular frames.
+  void scheduleWarmUpFrame({required VoidCallback beginFrame, required VoidCallback drawFrame}) {
```

Comment:
If this is meant for first frame only, why not call it `scheduleFirstFrame`?

Reply:
The biggest reason, in my opinion, is that this frame is not guaranteed to be rendered, for example if `Animator::BeginFrame` is not called before, or on Web (if you agree with [my other comment](https://github.com/flutter/engine/pull/50570#discussion_r1496722312)). Therefore this might not be the first frame.

Besides, I don't want to give developers the impression that scheduling the first frame using `scheduleFrame` is wrong. The `scheduleWarmUpFrame` really is for warming up. An app doesn't need warm-ups to perform correctly, but will perform better.

Reply:
In addition to what Tong said: The warmup frame is also used in hot reload scenarios where it isn't the first frame of the app.

Reply:
The semantics of this method sound pretty complicated. I guess warm up is fine then. Should we have some assertions in the engine for this method? For example, should we make sure it's only called for the first frame, and is never called after a `scheduleFrame` is called? Simply exposing this as a sibling to `scheduleFrame` seems to contain footguns.